xrpl/models/requests/unknown_request.py

```diff
@@ -0,0 +1,82 @@
+"""A generic request that can be used for unsupported requests."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Any, Dict, Type, Union, cast
+
+from xrpl.models.exceptions import XRPLModelException
+from xrpl.models.requests.request import Request, RequestMethod
+from xrpl.models.utils import require_kwargs_on_init
+
+
+@require_kwargs_on_init
+@dataclass(init=False, frozen=True)
+class UnknownRequest(Request):
+    """A request object representing all unsupported requests."""
+
+    method: RequestMethod = field(default=RequestMethod.UNKNOWN_REQUEST, init=False)
+    """
+    This field is required.
+
+    :meta hide-value:
+    """
+
+    def __init__(self: UnknownRequest, **kwargs: Dict[str, Any]) -> None:
+        """
+        Initializes an UnknownRequest.
+
+        Arguments:
+            kwargs: All the arguments for the request.
+        """
+        # initialize all the dataclass stuff
+        super().__init__(
+            id=(cast(Union[str, int, None], kwargs["id"]) if "id" in kwargs else None),
```

Comment:
you could take the `None` out of the Union

Reply:
I left it in in case someone specifically has `id=None` (which is theoretically possible).